File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/Vec2Uniform.java
package org.saar.lwjgl.opengl.shaders.uniforms;

import org.joml.Vector2fc;
import org.lwjgl.opengl.GL20;

public abstract class Vec2Uniform extends UniformBase implements Uniform {

    @Override
    public final void doLoad() {
        final Vector2fc v = getUniformValue();
        GL20.glUniform2f(getLocation(), v.x(), v.y());
    }

    public abstract Vector2fc getUniformValue();
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shader/uniforms/Vec2Uniform.kt
package org.saar.lwjgl.opengl.shader.uniforms

import org.joml.Vector2fc
import org.lwjgl.opengl.GL20

abstract class Vec2Uniform : Uniform {

    final override fun load(location: Int) {
        val value: Vector2fc = this.value
        GL20.glUniform2f(location, value.x(), value.y())
    }

    abstract val value: Vector2fc
}